packages/@aws-cdk/aws-route53/README.md

```diff
@@ -125,26 +125,45 @@ Constructs are available for A, AAAA, CAA, CNAME, MX, NS, SRV and TXT records.
 Use the `CaaAmazonRecord` construct to easily restrict certificate authorities
 allowed to issue certificates for a domain to Amazon only.
 
-To add a NS record to a HostedZone in different account
+To add a NS record to a HostedZone in different account you can do the following:
+
+In the account containing the parent hosted zone:
 
 ```ts
 import * as route53 from '@aws-cdk/aws-route53';
 
-// In the account containing the HostedZone
 const parentZone = new route53.PublicHostedZone(this, 'HostedZone', {
   zoneName: 'someexample.com',
-  crossAccountZoneDelegationPrincipal: new iam.AccountPrincipal('12345678901')
+  crossAccountZoneDelegationPrincipal: new iam.AccountPrincipal('12345678901'),
+  crossAccountZoneDelegationRoleName: 'MyDelegationRole',
```

Comment:
Is the primary intent of this new field to make it easier to pass the ARN between the parent/child account/stacks? Do you (or "we") have feedback that this is necessary? Would it make just as much sense to allow a user to pass in an `iam.IRole` here that is used instead of creating a new one?

Reply:
The ARN of the delegation role cannot be passed around to stacks sine they will have different environments (this is cross account delegation and CDK does not allow passing references across environments).

This means that users need to import the delegation role in the stack containing the child zone. Since it needs to be imported, the users would need the role ARN. Now this ARN can either be hard-coded or constructed. Hard-coding the ARN is not entirely feasible since that would require deploying the parent stack first, then making code change and then deploying child stack in pipelines.

With role name, the ARN can be easily constructed (refer readme for importing role) and it provides a better user experience.

Reply:
Another way could be to accept parentAccountId and delegationRoleName as props to the construct as well. But I though accepting role (does not matter where/how it got created) would be better.